### DISCUSSION TITLE
Backup using config and glob

### DISCUSSION THREAD

**Discussion by enboig** [rustic-rs/rustic#1194] [Q&A] [answered]
I am trying to make a backup with a single config file and snapshot; I think it can be achieved backing up `/`, removing everyghing and reading using glob, but it is not working:
```
[[backup.sources]]
source = "/"
glob = [
        "!/**",
        "/backup/**",
        "/etc/**",
        "/root/scripts/**",
        "/home/*/Documents/*",
        "!/home/*/Documents/temp/*",
       ]
```

Is it impossible? I know I could use multiple `[[backup.sources]]`, but it generates multiple snapshots.

**Comment by kapitainsky** [comment]
Here you are how it should be done

```
!*

/backup
/backup/**/*

/root
/root/scripts
/root/scripts/**/*

```

for Documents I am not sure but I think I would try:

```
/home
/home/**
/home/**/Documents
/home/**/Documents/**/*
!/home/**/Documents/temp/**/*
```

**Reply by enboig** [reply]
Thanks, this worked as expected.

**Comment by aawsome** [comment]
You can also use `source = ["/backup", "/etc", "/root/scripts", "/home"]` - note that the syntax changed; this is working with the nightly builds, rustic 0.7.0 has a different syntax.

For `/home`, however, you have to use globs to include/exclude everything unwanted.

**Comment by kapitainsky** [comment, marked as answer]
As I needed similar exclusions I did it and tested. For your folders please see working example.


I use glob file but of course you can include all in toml file if you want.

My toml file contains:

```
[backup]
glob-file = ["./rustic.glob"]

[[backup.sources]]
source = "/"
```

And here you are glob file content:
```
!*

/backup
/backup/**/*

/etc
/etc/**/*

/root
/root/scripts
/root/scripts/**/*

/home
/home/*/
/home/*/Documents
/home/*/Documents/**/*
!/home/*/Documents/temp
```

**Reply by enboig** [reply]
This also works. Thanks.

**Comment by intgr** [comment]
I had the same issue as OP here. I solved it by removing `one-file-system = true` from settings. It seems `one-file-system` does not work together with `source = "/"`.
